### PR TITLE
No datacubes view

### PR DESCRIPTION
**Pull request by lorem--ipsum**
Added the no-datacube view, removed the no-datacube application, and moved the header bars closer to their respective views.

### CODE REVIEW

File: src/client/views/no-data-view/no-data-view.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import { SvgIcon } from '../../components/index';
+
+import { NoDataHeaderBar} from './no-data-header-bar/no-data-header-bar';
+
+export interface NoDataViewProps extends React.Props<any> {
+  user?: User;
+  appSettings?: AppSettings;
+  onNavClick?: Fn;
+  onOpenAbout: Fn;
+  customization?: Customization;
```
Comment (vogievetsky) [inline src/client/views/no-data-view/no-data-view.tsx]: this should we aware of `stateful`

File: src/client/views/no-data-view/no-data-view.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+export interface NoDataViewState {
+  mode?: 'no-connected-data' | 'no-cube';
```
Comment (vogievetsky) [inline src/client/views/no-data-view/no-data-view.tsx]: please make this a typedef. i.e. `export type Mode = 'no-connected-data' | 'no-cube';` above

File: src/client/views/no-data-view/no-data-view.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    if (!user || !user.allow['settings']) return null;
+
+    return <div className="icon-button" onClick={this.goToSettings.bind(this)}>
+      <SvgIcon svg={require('../../icons/full-settings.svg')}/>
+    </div>;
+  }
+
+  renderTitle(mode: 'no-connected-data' | 'no-cube'): JSX.Element {
+    if (mode === 'no-cube') {
+      return <div className="title">
```
Comment (vogievetsky) [inline src/client/views/no-data-view/no-data-view.tsx]: This feels like a large amount of duplicated code, unless I am missing something the mode only affects the label.

File: src/client/views/no-data-view/no-data-view.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  renderSettingsIcon() {
+    const { user } = this.props;
+    if (!user || !user.allow['settings']) return null;
+
+    return <div className="icon-button" onClick={this.goToSettings.bind(this)}>
+      <SvgIcon svg={require('../../icons/full-settings.svg')}/>
+    </div>;
+  }
+
+  renderTitle(mode: 'no-connected-data' | 'no-cube'): JSX.Element {
```
Comment (vogievetsky) [inline src/client/views/no-data-view/no-data-view.tsx]: why not do `const { mode } = this.state` ?

File: src/client/views/no-data-view/no-data-view.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    }
+  }
+
+  goToSettings() {
+    window.location.hash = '#settings';
+  }
+
+  renderSettingsIcon() {
+    const { user } = this.props;
+    if (!user || !user.allow['settings']) return null;
```
Comment (vogievetsky) [inline src/client/views/no-data-view/no-data-view.tsx]: also check to make sure `stateful` is true

File: src/client/views/no-data-view/no-data-view.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+    return <div className="title">
+      <div className="icon">
+        <SvgIcon svg={require('../../icons/data-cubes.svg')}/>
+      </div>
+      <div className="label">{STRINGS.noConnectedData}</div>
+    </div>;
+  }
+
+  renderLink(mode: 'no-connected-data' | 'no-cube'): JSX.Element {
```
Comment (vogievetsky) [inline src/client/views/no-data-view/no-data-view.tsx]: check `stateful` here

**Comment by vogievetsky** [comment]
gr 🎱

File: src/client/views/no-data-view/no-data-view.tsx (hunk 2 above)
Comment (vogievetsky) [inline src/client/views/no-data-view/no-data-view.tsx]: also the modes should probably be something like `'no-cluster' | 'no-cube'`  I am not sure what no connected data means

File: src/client/views/no-data-view/no-data-view.tsx [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+  constructor() {
+    super();
+    this.state = {};
+  }
+
+  componentWillReceiveProps(nextProps: NoDataViewProps) {
+    const { clusters } = nextProps.appSettings;
+
+    if (!clusters || clusters.length) {
```
Comment (vogievetsky) [inline src/client/views/no-data-view/no-data-view.tsx]: I think this should be `(!clusters || !clusters.length)

File: src/client/views/no-data-view/no-data-view.tsx (hunk 4 above)
Comment (lorem--ipsum) [inline src/client/views/no-data-view/no-data-view.tsx]: Dunno, I like functions that don't depend on the context... But in this case I didn't really put that many thoughts in it ^^

**Pull request merged**